packages/material-ui-styles/src/styled/styled.d.ts

```diff
@@ -12,14 +12,16 @@ import * as React from 'react';
 export type ComponentCreator<Component extends React.ElementType> = <Theme, Props extends {} = any>(
   styles:
     | CreateCSSProperties<Props>
-    | ((props: { theme: Theme } & CoerceEmptyInterface<Props>) => CreateCSSProperties<Props>),
+    | (({ theme, ...props }: { theme: Theme } & Props) => CreateCSSProperties<Props>),
```

Comment:
Was this required? I think we just recently added it but only have a test for core/styles/styled.

Reply:
What exactly do you mean? the `CoerceEmptyInterface` if that is required, or the change itself?

Reply:
> That looks way better.
> 
> core/styles/styled needs the same treatment though.

I wonder if the duplication is really needed at all for the whole ComponentCreator, just to set a DefaultTheme in core/styles/styled. Or simply reuse the interface from styles in core/styles, adding a default to generic.

I also have to admit I have found some things on the way but its like a chain which leads to next, next, next thing and I wanted to make a working fix that is simple to merge and then think about improvements in general. Like the Theme generic in StyleOptions is for example is unused.

I had to factor out certain things to get something done without starting new and new and again new change :)

Reply:
> What exactly do you mean? the CoerceEmptyInterface if that is required, or the change itself?

The whole change of this line. It was only recently changed in core/styles/styled if I recall correctly. Should be more clear once the changes to styles/styled are applied to core/styles/styled.

Reply:
Looks like it is required, when I revert this change in core/styles/styled, test `/material-ui/test/typescript/styles.spec.tsx:572:5` fails

Reply:
> The whole change of this line. It was only recently changed in core/styles/styled if I recall correctly

@eps1lon https://github.com/mui-org/material-ui/pull/16356

Reply:
Change pushed

Reply:
@eps1lon change request still hanging, but does not look like the built-in change request feature on github. I applied the stuff you wanted to core/styles/styled. Do you also want the recent change undone? Because it then fails test, I tried it. If you still want to have it in, give me a feedback.

Reply:
I was just asking. If it fails without then I obviously don't want it.